### ISSUE TITLE
Add Support for Logfile Management

### ISSUE THREAD

**Issue by msymons** [DependencyTrack/dependency-track#242]
As an enhancement, it would be useful if Dependency-Track server implemented:

1.  Support for log roll-over.  Thus, individual logfiles would not grow too big, and it would not be necessary to stop the server in order to do things manually.
2.  Support for configuration of loglevel. eg,  the option of turning on logging at a finer level in order to locate the source of a problem.

Originally mentioned in #237

**Comment by stevespringett** [comment]
Dependency-Track (and Alpine) both use logback alongside slf4j. Logback is already capable of the features you describe.

Rollover (I agree) should be the default. So I'll leave this as an enhancement. But it can already be configured (via logback.xml). And logging level can also be configured independently of class or package, or for the application as a whole.

It may be easier (especially for Docker) to be able to specify an alternative location to this file. But you can specify an alternate location with `-Dlogback.configurationFile=/path/to/config.xml`

**Comment by msymons** [comment]
@stevespringett, forgive me for being a bit dense...  but, now that v3.4 has been released, is the rollover supposed to be configured in `application.properties`?  If so, how?

From the [changelog](https://docs.dependencytrack.org/changelog/)

> Added logfile rotation with default 10MB cap (configurable)

The default of 10MB explains why nothing has yet happened with my DT after upgrade from v3.3.1 to v3.4.0...  the logfile is only 7.7 MB in size!

However, my network connection to the server is very slow and I would like to config for rotation at something like 2MB, pending network upgrade.

By the way,  the changelog page on the documentation site has the word "rotation" but search for this word (using the site search function) returns zero hits.  Is changelog page not being indexed for search?

**Comment by stevespringett** [comment]
Refer to logback.xml or supply an alternative logging configuration via 

`-Dlogback.configurationFile=/path/to/config.xml`

The searching of change log entries can now be performed. Thanks for catching that.

**Comment by msymons** [comment]
I got things working... but it took a while.

I would like to suggest:

1.  Provide sample (default) `logback.xml` via documentation website (ie, in same way that a sample application.properties file is provided).  This is something that can then be copy and pasted to create a working config and then tweaked to suit.   eg, when you have a good starting point it becomes trivial to change (say) `maxFileSize`.
I found the default `logback.xml` in DT github repo.  It was only later that I had the "d'oh moment" and realized that it was also available inside the WAR.  It would have been a lot easier to just grab it from documentation!
2.  Perhaps also Include link(s) to relevent documentation on  qos website?
3. Extend application.properties to specify path to logging configuration, commented out by default.  This will cater for the situation where one has permissions to modify `application.properties` (and `logback.xml`) but not permissions to change service file or sysconfig file (which is where I currently specify the `logback.configurationFile` location.  It would also be a way of providing a line or two of documentation and act as a reminder that logfile configuration is possible.
